lib/rubygems/request_set/gem_dependency_api.rb

```diff
@@ -526,6 +552,7 @@ def gem_requires name, options # :nodoc:
     else
       @requires[name] << name
     end
+    raise "Unhandled gem options #{options.inspect}" unless options.empty?
```

Comment:
Hi @mpapis, I ran `ruby -Ilib bin/gem install --file` as suggested in https://github.com/rubygems/rubygems/pull/1341#issuecomment-141215685 and it still shows ``undefined method `version' for nil:NilClass``